src/test/java/com/kodilla/ecommercee/ProductTests.java

```diff
@@ -0,0 +1,136 @@
+package com.kodilla.ecommercee;
+
+import com.kodilla.ecommercee.domain.Product;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+
+@Transactional
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ProductTests {
+
+    @Autowired
+    private GenericEntityRepository genericEntityRepository;
+
+
+    @Test
+    public void testSaveProduct() {
+
+        //Given
+        Product product = new Product("Altana SunSet", "Altana ogrodowa ...", 999.90);
+
+        //When
+        genericEntityRepository.save(product);
+
+        //Then
+        Long id = product.getId();
+        Product foundProduct = (Product) genericEntityRepository.getOne(id);
+        Assert.assertNotEquals(null, foundProduct);
+
+        //CleanUp
+        genericEntityRepository.deleteById(id);
+
+    }
+
+    @Test
+    public void testUpdateProduct() {
+
+        //Given
+        Product product = new Product("Altana SunSet", "Altana ogrodowa ...", 999.90);
+        genericEntityRepository.save(product);
+
+        //When
```

Comment:
nie wiem czy  sekcja when,nie powinna przejść do sekcji given, a z sekcji then 2 pierwsze linijki do when

Reply:
@aleksandra4991 productToUpdate.setPrice(1000.73); to przedmiot testu więc zostawię w When